src/box/sql/build.c

```diff
@@ -115,6 +118,24 @@ encode_format(struct region *region, struct space_def *def, uint32_t *size)
 			mpstream_encode_str(&stream, "default");
 			mpstream_encode_str(&stream, default_str);
 		}
+		if (field->constraint_count == 0)
+			continue;
+		mpstream_encode_str(&stream, "foreign_key");
+		mpstream_encode_map(&stream, field->constraint_count);
+		for (uint32_t i = 0; i < field->constraint_count; ++i) {
+			struct tuple_constraint_def *cdef =
+				&field->constraint_def[i];
```

Comment:
Lets add an assertion that the type is `CONSTR_FKEY`. Same in `encode_opts()`.

Reply:
This part was moved to another PR. I added support for both `CONSTR_FUNC` and `CONSTR_FKEY`.